### ISSUE TITLE
BUG - Wrong URL in Updater for "Modding Tools" Domain Mods

### ISSUE THREAD

**Issue by ChemGuy1611** [Nexus-Mods/Vortex#16480]
**Describe the bug**
I noticed when attempting to update a Fluffy Mod Manager instance that one of my Vortex extensions downloads automatically, the update fails because the updater is looking for the mod in the game's domain, rather than in the Modding Tools domain. This occurs in spite of the fact that the mod details correctly identify that the mod came from the Modding Tools domain. Right-clicking on the mod to open the Nexus Mods page goes to the correct page for the mod. It seems that only the Updater is using the wrong URL. 

This bug may also occur for mods of this type that were not automatically downloaded. 

**To Reproduce**
1. Use an extension that auto-downloads a mod from the Modding Tools section (Dead Rising, RE4/7, SF6). 
2. Attempt to update the mod in Vortex.
3. Receive the error due to the incorrect URL. 

**Expected behavior**
Updater should check the Modding Tools domain page, rather than the game's domain. 

**Screenshots**
<img width="215" alt="Snap0112" src="https://github.com/user-attachments/assets/766e194a-2d5a-467c-bdb1-ce151a3ed689">

<img width="490" alt="Snap0113" src="https://github.com/user-attachments/assets/d40b8e15-28d9-4f1b-a7f5-faa956a2c2c9">

**Platform (please complete the following information):**
 - OS: Windows 11
 - Vortex Version 1.12.4

**Comment by IDCs** [comment]
Hey there @ChemGuy1611, the error there is misleading - the reason why you were unable to update FMM is down to the fact that Fluffy did not build the update chain correctly when he uploaded a new version. Simply put, Vortex knows that there is a new version, but it has no idea what it is, or where it's located. There's a fallback mechanism which will default to the active gameId if the 'site' domain does not hold that information.

That is unfortunately a workaround to allow support for multiple compatible domain ids (e.g. Skyrim and SkyrimVR or FO4 and FOLON) and I'm afraid there's little I can change without breaking support for game extensions that are actively using the fallback.

Simply put - the Nexus API data is unfortunately not sufficient for us to make a decision when a file has multiple compatible domains and the update chain is not provided by the author of the file.

I'm afraid that this won't be fixed until the API team provides us with the required data (will not happen in Vortex's lifetime)

**Comment by IDCs** [comment]
@ChemGuy1611, no worries - re-opening!

I'll run some tests using your FSR mod and let you know what I find - I'm afraid there's currently no way to force a specific domain when checking for updates as a user.

**Comment by IDCs** [comment]
Hey @ChemGuy1611, I can't seem to reproduce this issue on Vortex's main branch. We're planning to release a 1.14 beta this week and would be good if you could test this out and let us know if you're still encountering it. Are you perhaps setting the compatible games in your downloader code?

Either way, I'll let you know once 1.14 is built and will probably send you a download link ahead of us releasing it.

**Comment by IDCs** [comment]
Hey @ChemGuy1611 - as promised here's a link to the 1.14 beta artifact https://github.com/Nexus-Mods/Vortex/actions/runs/13785124728/artifacts/2728844507

Let me know if you're still encountering the issue.

Out of the game extensions you developed and are setting the domain to "site" - can you send me the name of one so I can run some more tests.

**Comment by ChemGuy1611** [comment]
You can use Dragon's Dogma 2 or Monster Hunter Wilds (Fluffy MM).Or the Like a Dragon Pirate Yakuza in Hawaii (Shin Ryu MM, though we are changing that to a GitHub download).

**Comment by ChemGuy1611** [comment]
@IDCs - I tried on the 1.14 beta, and I did not see a change in this behavior. 

Another thing I've discovered is that it does not make a difference in the auto-downloader functions if I set the game in the dlinfo to the game domain or to 'site'. As far as the update checking is concerned. Vortex automatically updates the metadata after install to 'site' domain. This makes sense as the error is also seen when installing mods directly from the site to Vortex, such as FSR or dlssg-to-fsr3.

**Comment by IDCs** [comment]
@ChemGuy1611 - ok managed to reproduce and fix it using DD2.

The underlying issue is the fact that the `game.details.nexusPageId` property was defined - that legacy property was initially developed as a way to allow users to download mods that are meant for another game, it completely overrides the domain name. nexusPageId has since been superseded by the compatibleDownloads property.

The compatibleDownloads property combined with the improved download metadata we developed pretty much makes nexusPageId obsolete, yet it's still kept in there to support older game extensions.

In the short-term, just remove the nexusPageId property from your game spec to allow the downloadGameId metadata to work out the domain name.

In the long-term, the new metadata will have priority over nexusPageId, but this fix will have to go through the regular beta QA/release process which means it may take a while for it to be released to stable.

**Comment by ChemGuy1611** [comment]
@IDCs - Everything is working now after I removed the nexusPageId. This has been bothering me for months. Thanks for the solution!
